biotrainer/trainers/trainer.py

```diff
@@ -173,6 +177,14 @@ def _create_and_load_embeddings(self) -> Dict[str, Any]:
 
         # Mapping from id to embeddings
         id2emb = embedding_service.load_embeddings(embeddings_file_path=embeddings_file)
+        if (self._dimension_reduction_method and 
+            self._reduced_dimension and
+            len(id2emb)>=3 and
+            list(id2emb.values())[0].shape[0]>=3):
```

Comment:
I would suggest moving this to a separate function to check the condition, and maybe also include another check for the protocol here.